adapter/internal/oasparser/model/mgw_swagger.go

```diff
@@ -838,14 +838,17 @@ func ResolveThrottlingTier(vendorExtensions map[string]interface{}) string {
 // x-wso2-disable-security : true/false to enable and disable security.
 func ResolveDisableSecurity(vendorExtensions map[string]interface{}) bool {
 	disableSecurity := false
+	isXWso2DisableSecurityAvailable := false
 	y, vExtAuthType := vendorExtensions[xAuthType]
 	z, vExtDisableSecurity := vendorExtensions[xWso2DisableSecurity]
 	if vExtDisableSecurity {
 		// If x-wso2-disable-security is present, then disableSecurity = val
 		if val, ok := z.(bool); ok {
 			disableSecurity = val
+			isXWso2DisableSecurityAvailable = val
 		}
-	} else if vExtAuthType {
+	} 
+	if vExtAuthType && !isXWso2DisableSecurityAvailable {
```

Comment:
isnt it always isXWso2DisableSecurityAvailable and disableSecurity is same till this line. So shall we use the disableSecurity here